azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_pipeline.py

```diff
@@ -155,26 +156,66 @@ def _on_disconnected():
             if self.on_disconnected:
                 self.on_disconnected()
 
+        # Set internal event handlers
         self._pipeline.on_pipeline_event_handler = _on_pipeline_event
         self._pipeline.on_connected_handler = _on_connected
         self._pipeline.on_disconnected_handler = _on_disconnected
 
+        # Initialize the pipeline
         callback = EventedCallback()
-
-        # NOTE: It would be nice if this didn't have to go down as a dynamic operation.
-        # At this time we haven't been able to figure out a better way to make it work though.
-
         op = pipeline_ops_base.InitializePipelineOperation(callback=callback)
-
         self._pipeline.run_op(op)
         callback.wait_for_completion()
 
+        # Set the running flag
+        self._running = True
+
+    def _ensure_running(self):
```

Comment:
>_ensure_running [](start = 8, length = 15)

This is different from all the other times I've seen/used "ensure" as a verb.  I would assume that "ensure running" means "if it's not running, make it run" and you're using it to mean "if it's not running, fail".  (This is the same issue as the "make sure the transport is disconnected" comment that I made elsewhere.)

Reply:
I'll change this to "verfiy_running" instead, there's less room for confusion of meaning there